--- a/conf/nginx.conf
+++ b/conf/nginx.conf
@@ -1,18 +1,38 @@
 
-#user  nobody;
-worker_processes  1;
+user www www;
+worker_processes  2;
 
-#error_log  logs/error.log;
+error_log   /var/log/nginx_error.log;
 #error_log  logs/error.log  notice;
 #error_log  logs/error.log  info;
 
-#pid        logs/nginx.pid;
+pid        /var/run/nginx.pid;
 
+thread_pool io_pool threads=8;
 
 events {
+    use epoll;
     worker_connections  1024;
 }
 
+#rtmp配置：
+rtmp {
+    server {
+        listen 1935;
+        application live {
+            live on;
+            #record off;
+        }
+        #增加对HLS支持开始
+        application hls {
+            live on;
+            hls on;
+            hls_path /srv/www/hls;
+            hls_fragment 5s; 
+        }
+        #增加对HLS支持结束
+    }
+}
 
 http {
     include       mime.types;
@@ -22,63 +42,70 @@ http {
     #                  '$status $body_bytes_sent "$http_referer" '
     #                  '"$http_user_agent" "$http_x_forwarded_for"';
 
-    #access_log  logs/access.log  main;
+    access_log  /var/log/nginx_access.log ;
 
     sendfile        on;
-    #tcp_nopush     on;
+    tcp_nopush     on;
 
     #keepalive_timeout  0;
     keepalive_timeout  65;
 
-    #gzip  on;
+    gzip  on;
+    #include       vhost.conf;
 
+    ###########################owncloud配置###########################
+    #include       owncloud.conf;
+    #################################################################
     server {
-        listen       80;
+        listen       88;
         server_name  localhost;
-
-        #charset koi8-r;
-
+        #error_log /var/log/nginx/error_debug.log debug;
+        #charset UTF-8;
         #access_log  logs/host.access.log  main;
-
+        #autoindex on;
+        #autoindex_exact_size off;
+        #autoindex_localtime on;
         location / {
-            root   html;
-            index  index.html index.htm;
+            root   /srv/www;
+            index  index.php index.html index.htm;
+#            aio        threads=io_pool;
+#           auth_basic           "closed site";
+#          auth_basic_user_file conf/pwd;
+#	   random_index on;
         }
-
+        #HLS配置开始,这个配置为了`客户端`能够以http协议获取HLS的拉流
+        location /hls {
+            # Serve HLS fragments
+            autoindex on;
+            autoindex_exact_size off;
+            autoindex_localtime on;
+            types {
+                application/vnd.apple.mpegurl m3u8;
+                video/mp2t ts;
+            }
+            root /srv/www;
+            add_header Cache-Control no-cache;
+        }
+       #HLS配置结束
+       
         #error_page  404              /404.html;
-
-        # redirect server error pages to the static page /50x.html
-        #
         error_page   500 502 503 504  /50x.html;
         location = /50x.html {
             root   html;
         }
-
-        # proxy the PHP scripts to Apache listening on 127.0.0.1:80
-        #
-        #location ~ \.php$ {
-        #    proxy_pass   http://127.0.0.1;
-        #}
-
-        # pass the PHP scripts to FastCGI server listening on 127.0.0.1:9000
-        #
-        #location ~ \.php$ {
-        #    root           html;
-        #    fastcgi_pass   127.0.0.1:9000;
-        #    fastcgi_index  index.php;
-        #    fastcgi_param  SCRIPT_FILENAME  /scripts$fastcgi_script_name;
-        #    include        fastcgi_params;
-        #}
-
-        # deny access to .htaccess files, if Apache's document root
-        # concurs with nginx's one
-        #
-        #location ~ /\.ht {
-        #    deny  all;
-        #}
+        location ~ \.php$ {
+            try_files $uri =404;
+            root           /srv/www;
+            fastcgi_pass   127.0.0.1:9000;
+            fastcgi_index  index.php;
+            #fastcgi_param  SCRIPT_FILENAME  /scripts$fastcgi_script_name;
+            fastcgi_param  SCRIPT_FILENAME  $document_root$fastcgi_script_name;
+            include        fastcgi_params;
+        }
     }
 
 
+
     # another virtual host using mix of IP-, name-, and port-based configuration
     #
     #server {
@@ -96,17 +123,18 @@ http {
     # HTTPS server
     #
     #server {
-    #    listen       443 ssl;
+    #    listen       443;
     #    server_name  localhost;
 
+    #    ssl                  on;
     #    ssl_certificate      cert.pem;
     #    ssl_certificate_key  cert.key;
 
-    #    ssl_session_cache    shared:SSL:1m;
     #    ssl_session_timeout  5m;
 
+    #    ssl_protocols  SSLv2 SSLv3 TLSv1;
     #    ssl_ciphers  HIGH:!aNULL:!MD5;
-    #    ssl_prefer_server_ciphers  on;
+    #    ssl_prefer_server_ciphers   on;
 
     #    location / {
     #        root   html;
